script/uninstall.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env sh
+set -eu
+
+# Uninstalls Zed that was installed using the install.sh script
+# Modeled after install.sh
+
+main() {
+    platform="$(uname -s)"
+    channel="${ZED_CHANNEL:-stable}"
+
+    if [ "$platform" = "Darwin" ]; then
+        platform="macos"
+    elif [ "$platform" = "Linux" ]; then
+        platform="linux"
+    else
+        echo "Unsupported platform $platform"
+        exit 1
+    fi
+
+    "$platform"
+
+    echo "Zed has been uninstalled"
+}
+
+linux() {
+    suffix=""
+    if [ "$channel" != "stable" ]; then
+        suffix="-$channel"
+    fi
+
+    appid=""
+    case "$channel" in
+      stable)
+        appid="dev.zed.Zed"
+        ;;
+      nightly)
+        appid="dev.zed.Zed-Nightly"
+        ;;
+      preview)
+        appid="dev.zed.Zed-Preview"
+        ;;
+      dev)
+        appid="dev.zed.Zed-Dev"
+        ;;
+      *)
+        echo "Unknown release channel: ${channel}. Using stable app ID."
+        appid="dev.zed.Zed"
+        ;;
+    esac
+
+    # Remove the app directory
+    rm -rf "$HOME/.local/zed$suffix.app"
+
+    # Remove the binary symlink
+    rm -f "$HOME/.local/bin/zed"
+
+    # Remove the .desktop file
+    rm -f "$HOME/.local/share/applications/${appid}.desktop"
+}
+
+macos() {
+    app="Zed.app"
+    case "$channel" in
+      nightly)
+        app="Zed Nightly.app"
+        ;;
+      preview)
+        app="Zed Preview.app"
+        ;;
+      dev)
+        app="Zed Dev.app"
+        ;;
+    esac
+
+    # Remove the app bundle
+    if [ -d "/Applications/$app" ]; then
+        rm -rf "/Applications/$app"
+    fi
+
+    # Remove the binary symlink
+    rm -f "$HOME/.local/bin/zed"
```

Comment:
I think we would also need to deal with the `~/Library/Application\ Support/Zed` directory, but it's a bit trickier, as multiple Zed versions may store things there:

```
Library/Application Support/Zed ❯ l .
total 16
drwxr-xr-x@  12 someonetoignore  staff   384B Oct  3 17:38 ./
drwx------+ 169 someonetoignore  staff   5.3K Nov 23 02:16 ../
-rw-r--r--@   1 someonetoignore  staff   6.0K Mar 26  2024 .DS_Store
drwxr-xr-x    3 someonetoignore  staff    96B Jul 13 03:12 copilot/
drwxr-xr-x    7 someonetoignore  staff   224B Sep 19 22:23 db/
drwxr-xr-x    3 someonetoignore  staff    96B Jun 14 01:37 docs/
drwxr-xr-x    6 someonetoignore  staff   192B Nov  6 01:15 extensions/
drwxr-xr-x   14 someonetoignore  staff   448B Nov 16 18:02 languages/
drwxr-xr-x@   6 someonetoignore  staff   192B Sep 24 23:20 node/
drwxr-xr-x    6 someonetoignore  staff   192B May 30 17:54 prettier/
drwxr-xr-x    3 someonetoignore  staff    96B Oct  3 17:38 remote_servers/
drwxr-xr-x    3 someonetoignore  staff    96B Jul 24 02:25 supermaven/
```

Technically, removing most of these is fine, as other Zed versions will redownload the needed things.
It's not great, esp. if the person is behind the VPN or offline, but seems like a better thing that not removing anything at all (as usually, there's only one version Zed users have installed).

The only notable thing is the `db` dir:
```
Library/Application Support/Zed ❯ l db
total 0
drwxr-xr-x   7 someonetoignore  staff   224B Sep 19 22:23 ./
drwxr-xr-x@ 12 someonetoignore  staff   384B Oct  3 17:38 ../
drwxr-xr-x   5 someonetoignore  staff   160B May 30 18:27 0-dev/
drwxr-xr-x   5 someonetoignore  staff   160B Sep 19 22:23 0-global/
drwxr-xr-x   5 someonetoignore  staff   160B May 30 17:54 0-nightly/
drwxr-xr-x   5 someonetoignore  staff   160B May 30 21:23 0-preview/
drwxr-xr-x   5 someonetoignore  staff   160B May 30 21:23 0-stable/
```

we should only remove the one corresponding to the current version, and only remove `db` if it's empty afterwards.

-------------

Similar thing has to be done for Linux, it's in `~/.local/share/zed/` and besides all that have `zed-$VERSION.sock` files that have to follow the same logic.

Reply:
What I wanna do is the following:
```bash
db_dir="$HOME/Library/Application Support/Zed/db"
if [ -d "$db_dir/0-$db_suffix" ]; then
    rm -rf "$db_dir/0-$db_suffix"
    # If only global db remains, remove entire Zed directory
    if [ "$(ls -1 "$db_dir" | grep -v global | wc -l)" -eq 0 ]; then
        rm -rf "$HOME/Library/Application Support/Zed"
    fi
fi
```
So, by using database as an indicator, we can pretty accurately determine whether there are any more installations left

Reply:
Also, what is the .sock file name? Is it just `zed-{nightly,preview,stable}.sock`? Is it in the ~/.local/share/zed?

Reply:
Yes to both.